src/ICSharpCode.SharpZipLib/Encryption/ZipAESTransform.cs

```diff
@@ -88,6 +89,23 @@ public ZipAESTransform(string key, byte[] saltBytes, int blockSize, bool writeMo
 			_writeMode = writeMode;
 		}
 
+		/// <summary>
+		/// Append all of the last transformed input data to the HMAC.
+		/// </summary>
+		public void AppendAllPending()
+		{
+			_appendHmac(0);
+		}
+		
+		/// <summary>
+		/// Append all except the number of bytes specified by remaining of the last transformed input data to the HMAC.
+		/// </summary>
+		/// <param name="remaining">The number of bytes not to be added to the HMAC. The excluded bytes are form the end.</param>
+		public void AppendFinal(int remaining)
+		{
+			_appendHmac(remaining);
```

Comment:
This delegate is only set if in Manual mode, otherwise it's just a nop, right? Why not just bail inside the method if not `ManualHmac` is set?
It just seems like added complexity, or am I missing something?

Reply:
Yes you are missing that the delegate captures the information when
   TransformBlock(byte[] inputBuffer, int inputOffset, int inputCount, byte[] outputBuffer, int outputOffset)
       _appendHmac = remaining => _hmacsha1.AppendData(inputBuffer, inputOffset, inputCount - remaining);
is called. During AppendFinal and AppendAllPending that information is not available anymore.